src/Rector/Deprecation/ExtensionPathRector.php

```diff
@@ -0,0 +1,140 @@
+<?php declare(strict_types=1);
+
+namespace DrupalRector\Rector\Deprecation;
+
+use DrupalRector\Rector\ValueObject\ExtensionPathConfiguration;
+use DrupalRector\Utility\AddCommentTrait;
+use PhpParser\Node;
+use Rector\Core\Contract\Rector\ConfigurableRectorInterface;
+use Rector\Core\Rector\AbstractRector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\ConfiguredCodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+class ExtensionPathRector extends AbstractRector implements ConfigurableRectorInterface
+{
+    use AddCommentTrait;
+
+    /**
+     * @var ExtensionPathConfiguration[]
+     */
+    private array $configuration;
+
+    public function configure(array $configuration): void
+    {
+        $this->configureNoticesAsComments($configuration);
+
+        foreach ($configuration as $value) {
+            if (!($value instanceof ExtensionPathConfiguration)) {
+                throw new \InvalidArgumentException(sprintf(
+                    'Each configuration item must be an instance of "%s"',
+                    ExtensionPathConfiguration::class
+                ));
+            }
+        }
+
+        $this->configuration = $configuration;
+    }
+
+    public function getNodeTypes(): array
+    {
+        return [
+            Node\Stmt\Expression::class,
+        ];
+    }
+
+    public function refactor(Node $node): ?Node
+    {
+        assert($node instanceof Node\Stmt\Expression);
+        if (!($node->expr instanceof Node\Expr\FuncCall)) {
+            return null;
+        }
+
+        foreach ($this->configuration as $configuration) {
+            $expr = $node->expr;
+            if ($this->getName($expr->name) !== $configuration->getFunctionName()) {
+                continue;
+            }
+            $args = $expr->getArgs();
+            if (count($args) !== 2) {
+                $this->addDrupalRectorComment($node, "Invalid call to {$configuration->getFunctionName()}, cannot process.");
+                return $node;
+            }
+            [$extensionType, $extensionName] = $args;
+
+            $extensionTypeValueType = $this->nodeTypeResolver->getType($extensionType->value);
+            if ($extensionTypeValueType->getConstantStrings()) {
+                $extensionType = $extensionTypeValueType->getValue();
+            }
+
+            if (in_array($extensionType, [
```

Comment:
Is it possible `$extensionType` is not a string here? So `!is_string` in case the constant string value couldn't be detected, such as PHPStan considering it as a union (I don't know how or why.)

Reply:
I don't know, i've not audited the code in the Rectors. There might be an edge case imaginable, although i haven't really seen any feedback on this change in any of the regular channels.

When searching gitlab i see this comment nowhere, but since updatebot doesnt do comments this might not mean much.

Reply:
Why not? Needle can be mixed and we check typesafe. So why error? We handle the non-string value "perfectly".

PHPStan is ran on this code.

Reply:
We're at PHPStan level 1

```
parameters:
	level: 1
```

We can merge as is. It'd probably never error, but it could. We can fix later once PHPStan is bumped up

Reply:
Ah, oops. Yeah that is not very helpfull then :x